WalletWasabi/Models/TransactionBuilding/FeeStrategy.cs

```diff
@@ -0,0 +1,79 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Models.TransactionBuilding
+{
+	public class FeeStrategy
+	{
+		private int _target;
+		private FeeRate _rate;
+
+		public FeeStrategyType Type { get; }
+
+		public int Target
+		{
+			get
+			{
+				if (Type != FeeStrategyType.Target)
+				{
+					throw new NotSupportedException($"Cannot get {nameof(Target)} with {nameof(FeeStrategyType)} {Type}.");
+				}
+				return _target;
+			}
+		}
+
+		public FeeRate Rate
+		{
+			get
+			{
+				if (Type != FeeStrategyType.Rate)
+				{
+					throw new NotSupportedException($"Cannot get {nameof(Rate)} with {nameof(FeeStrategyType)} {Type}.");
+				}
+				return _rate;
+			}
+		}
+
+		public static FeeStrategy TwentyMinutesConfirmationTargetStrategy { get; } = CreateFromConfirmationTarget(Constants.TwentyMinutesConfirmationTarget);
+		public static FeeStrategy OneDayConfirmationTargetStrategy { get; } = CreateFromConfirmationTarget(Constants.OneDayConfirmationTarget);
+		public static FeeStrategy SevenDaysConfirmationTargetStrategy { get; } = CreateFromConfirmationTarget(Constants.SevenDaysConfirmationTarget);
+
+		public static FeeStrategy CreateFromConfirmationTarget(int confirmationTarget) => new FeeStrategy(FeeStrategyType.Target, confirmationTarget: confirmationTarget, feeRate: null);
+
+		public static FeeStrategy CreateFromFeeRate(FeeRate feeRate) => new FeeStrategy(FeeStrategyType.Rate, confirmationTarget: null, feeRate: feeRate);
+
+		private FeeStrategy(FeeStrategyType type, int? confirmationTarget, FeeRate feeRate)
+		{
+			Type = type;
+			if (type == FeeStrategyType.Rate)
+			{
+				if (confirmationTarget != null)
+				{
+					throw new ArgumentException($"{nameof(confirmationTarget)} must be null.");
+				}
+				Guard.NotNull(nameof(feeRate), feeRate);
+				if (feeRate < new FeeRate(1m))
+				{
+					throw new ArgumentOutOfRangeException(nameof(feeRate), feeRate, "Vannot be less than 1s/b.");
```

Comment:
Vannot -> `Cannot`
Isn't it better to write 1 satoshi/byte? if not the add space after 1.

Reply:
Usually you throw this exception like this:
throw new ArgumentOutOfRangeException($"{nameof(feeRate)} cannot be less than 1 s/b.");

Reply:
Fixed the exception message with your first suggestion. However your second suggestion is unnecessary in this case, because you can feed `ArgumentOutOfRangeException` with the name of the variable and with the value of it.

![image](https://user-images.githubusercontent.com/9156103/63220737-257d5b80-c18e-11e9-9037-0cb6bbd9b34d.png)

Reply:
@nopara73 Maybe it is better to use `sat/byte` because it is used elsewhere.

Reply:
Done.